### PR TITLE
Use java.util.Objects to implement equals, hashCode

### PR DESCRIPTION
**Pull request by dagnir**
## Description
Use `java.util.Objects` to implements `equals()` and `hashCode()`.

## Motivation and Context
Should cut down on generated file size.

## Testing
Updated reference tests.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the **CONTRIBUTING** document
- [x] Local run of `mvn install` succeeds
- [x] My code follows the code style of this project
- [ ] My change requires a change to the Javadoc documentation
- [ ] I have updated the Javadoc documentation accordingly
- [x] I have read the **README** document
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

**Comment by millems** [comment]
This creates a bit more garbage because it needs to load things into an array. I'm guessing we're balancing against jar size? How much does this help that?

**Comment by dagnir** [comment]
This uses `Objects.hashCode` rather than `Objects.hash`, so don't think the array creation is an issue here

**Comment by millems** [comment]
Cool, I missed that!

File: codegen/src/test/resources/software/amazon/awssdk/codegen/poet/model/alltypesresponse.java [hunk 1]
```diff
@@ -494,218 +494,131 @@ public static Builder builder() {
...
+    AllTypesResponse other = (AllTypesResponse) obj;
+    if (!Objects.equals(stringMember(), other.stringMember()) {
+      return false;
+    }
+    if (!Objects.equals(integerMember(), other.integerMember()) {
+      return false;
+    }
+    if (!Objects.equals(booleanMember(), other.booleanMember()) {
+      return false;
+    }
```
Comment (millems) [inline codegen/src/test/resources/software/amazon/awssdk/codegen/poet/model/alltypesresponse.java]: Any chance we could just do return Object.equals() && Object.equals() & ...?

Comment (dagnir) [reply codegen/src/test/resources/software/amazon/awssdk/codegen/poet/model/alltypesresponse.java]: +1 sounds good

**Comment by dagnir** [comment]
Updated `equals` generation to chain `&&`

File: codegen/src/main/java/software/amazon/awssdk/codegen/poet/model/ModelMethodOverrides.java [hunk 2]
```diff
@@ -55,21 +62,25 @@ public MethodSpec equalsMethod(ShapeModel shapeModel) {
...
+                final String getterName = m.getFluentGetterMethodName();
+                memberEqualsStmt.add(sep)
+                        .add(" ")
+                        .add("$T.equals($N(), other.$N())", Objects.class, getterName, getterName);
+            };
+            memberModels.stream().findFirst().ifPresent(m -> adder.accept("return", m));
+            memberModels.stream()
+                    .skip(1)
+                    .forEach(m -> adder.accept("&&", m));
+            memberEqualsStmt.add(";");
```
Comment (millems) [inline codegen/src/main/java/software/amazon/awssdk/codegen/poet/model/ModelMethodOverrides.java]: Does the [PoetCollectors.toDelimitedCodeBlock](https://github.com/aws/aws-sdk-java-v2/blob/master/codegen/src/main/java/software/amazon/awssdk/codegen/poet/PoetCollectors.java#L41) make this a bit easier to implement?

Comment (dagnir) [reply codegen/src/main/java/software/amazon/awssdk/codegen/poet/model/ModelMethodOverrides.java]: Nice! Didn't know that was there. Done

**Pull request merged**